xsensTFBridge/src/xsensTFBridge.cpp

```diff
@@ -0,0 +1,257 @@
+        tf.transforms[24].child_frame_id = "L3_f1";
+        tf.transforms[25].child_frame_id = "T12_f1";
+        tf.transforms[26].child_frame_id = "T8_f1";
+        tf.transforms[27].child_frame_id = "T8_f2";
+        tf.transforms[28].child_frame_id = "Neck_f1";
+        tf.transforms[29].child_frame_id = "Neck_f2";
+        tf.transforms[30].child_frame_id = "Head_f1";
+        tf.transforms[31].child_frame_id = "RightUpperArm_f1";
+        tf.transforms[32].child_frame_id = "RightUpperArm_f2";
+        tf.transforms[33].child_frame_id = "RightForeArm_f1";
+        tf.transforms[34].child_frame_id = "RightHand_f1";
+        tf.transforms[35].child_frame_id = "LeftUpperArm_f1";
+        tf.transforms[36].child_frame_id = "LeftUpperArm_f2";
+        tf.transforms[37].child_frame_id = "LeftForeArm_f1";
+        tf.transforms[38].child_frame_id = "LeftHand_f1";
+        tf.transforms[39].child_frame_id = "RightUpperLeg_f1";
+        tf.transforms[40].child_frame_id = "RightUpperLeg_f2";
+        tf.transforms[41].child_frame_id = "RightLowerLeg_f1";
+        tf.transforms[42].child_frame_id = "RightFoot_f1";
+        tf.transforms[43].child_frame_id = "RightFoot_f2";
+        tf.transforms[44].child_frame_id = "LeftUpperLeg_f1";
+        tf.transforms[45].child_frame_id = "LeftUpperLeg_f2";
+        tf.transforms[46].child_frame_id = "LeftLowerLeg_f1";
+        tf.transforms[47].child_frame_id = "LeftFoot_f1";
+        tf.transforms[48].child_frame_id = "LeftFoot_f2";
+       
+        for (i=trueSegmentsCount;i<(segmentsCount);i++){
+            tf.transforms[i].header.seq   = i;
+            tf.transforms[i].transform.translation.x = 0;
+            tf.transforms[i].transform.translation.y = 0;
+            tf.transforms[i].transform.translation.z = 0;
+            tf.transforms[i].transform.rotation.x = 0;
+            tf.transforms[i].transform.rotation.y = 0;
+            tf.transforms[i].transform.rotation.z = 0;
+            tf.transforms[i].transform.rotation.w = 1;
+          }
+        
+        xsensDataPort.useCallback(*this);
+        //std::string xsensServerName = "/" + rf.find("serverName").asString() + "/frames:o";
+        std::string xsensServerName = "/" + serverName + "/frames:o";
+        std::string frameReaderPortName = "/" + getName() + "/frames:i";
+        xsensDataPort.open(frameReaderPortName);           
+        if (!Network::connect(xsensServerName.c_str(),frameReaderPortName))
+        {
+            std::cout << "Error! Could not connect to server " << xsensServerName << std::endl;
+            return false;
+        }
+
+        return true;
+    }
+    // Interrupt function.
+    bool interruptModule()
+    {
+        cout << "Interrupting your module, for port cleanup" << endl;
+        return true;
+    }
+    // Close function, to perform cleanup.
+    bool close()
+    {
+        // optional, close port explicitly
+        cout << "Calling close function\n";
+        return true;
```

Comment:
Here you should actually cleanup your module.
You should close the ports that you opened in the `configure`